vcd/resource_vcd_vdc.go

```diff
@@ -0,0 +1,499 @@
+						},
+					},
+				},
+				Description: "Storage profiles supported by this VDC.",
+			},
+			"resource_guaranteed_memory": &schema.Schema{
+				Type:        schema.TypeFloat,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Percentage of allocated memory resources guaranteed to vApps deployed in this VDC. For example, if this value is 0.75, then 75% of allocated resources are guaranteed. Required when AllocationModel is AllocationVApp or AllocationPool. Value defaults to 1.0 if the element is empty.",
+			},
+			"resource_guaranteed_cpu": &schema.Schema{
+				Type:        schema.TypeFloat,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Percentage of allocated CPU resources guaranteed to vApps deployed in this VDC. For example, if this value is 0.75, then 75% of allocated resources are guaranteed. Required when AllocationModel is AllocationVApp or AllocationPool. Value defaults to 1.0 if the element is empty.",
+			},
+			"v_cpu_in_mhz": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Specifies the clock frequency, in Megahertz, for any virtual CPU that is allocated to a VM. A VM with 2 vCPUs will consume twice as much of this value. Ignored for ReservationPool. Required when AllocationModel is AllocationVApp or AllocationPool, and may not be less than 256 MHz. Defaults to 1000 MHz if the element is empty or missing.",
+			},
+			"is_thin_provision": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Boolean to request thin provisioning. Request will be honored only if the underlying datastore supports it. Thin provisioning saves storage space by committing it on demand. This allows over-allocation of storage.",
+			},
+			"network_pool_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "A name of a network pool in the Provider VDC. Required if this VDC will contain routed or isolated networks.",
+			},
+			"provider_vdc_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "A reference to the Provider VDC from which this organization VDC is provisioned.",
+			},
+			"uses_fast_provisioning": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Boolean to request fast provisioning. Request will be honored only if the underlying datastore supports it. Fast provisioning can reduce the time it takes to create virtual machines by using vSphere linked clones. If you disable fast provisioning, all provisioning operations will result in full clones.",
+			},
+			"over_commit_allowed": &schema.Schema{
```

Comment:
I have a problem with the semantics here:

"allowed" in the property name is a positive value
"Set false to disallow" in the description is a double negative that doesn't match the meaning of the name.
Moreover, an optional boolean property is false by default. How do we set it to false? By leaving it blank? But that is a silent specification, which we don't want to allow.
I think we need to rename the property in a way that reflects the action we can perform, such as "over_commit_disallowed" meaning that by default it is allowed, and if we set this property we will disallow it.

Reply:
I find the suggested "over_commit_disallowed" more confusing that what we have here now, because then you do have to play with double inverse logic. It's always more confusing when the parameter has "not" ("dis" in this case) in the name.

Description says that "Defaults to true if empty or missing". Is this actual behaviour? If it is, I suggest renaming to `allow_over_commit`, which is simpler and consistent with `allow_(thin/fast)_provisioning` and if someone wants to disallow this behaviour, they will set `allow_over_commit = "false"`.

Looking from a wider perspective, is there such parameter in the GUI (I can't find it)? What is the added value of having it and how come the GUI chose not to expose it?

Thanks!

Reply:
Changed to allow_over_commit and default value true. Comment is improved.